src/NServiceBus.Callbacks/RequestResponseExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+
+    /// <summary>
+    /// Synchronous request/response extension methods.
+    /// </summary>
+    public static class RequestResponseExtensions
+    {
+        /// <summary>
+        /// Sends a <paramref name="requestMessage"/> to the configured destination and returns back a <see cref="SendContext{TResponse}"/> that allows the use user to wait on.
+        /// </summary>
+        /// <typeparam name="TResponse">The response type.</typeparam>
+        /// <param name="bus">Object beeing extended.</param>
+        /// <param name="requestMessage">The request message.</param>
+        /// <param name="options">The options for the send.</param>
+        /// <returns>A synchronous request/response context.</returns>
+        public static SendContext<TResponse> SynchronousRequestResponse<TResponse>(this IBus bus, object requestMessage, SynchronousOptions options)
+        {
+            if (requestMessage == null)
+            {
+                throw new ArgumentNullException("requestMessage");
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException("options");
+            }
+
+            if (bus == null)
+            {
+                throw new ArgumentNullException("bus");
+            }
+
+            var sendOptions = new SendOptions(options.Destination, options.CorrelationId);
+                
+            foreach (var header in options.Headers)
+            {
+                sendOptions.AddHeader(header.Key, header.Value);
+            }
+
+            sendOptions
+                .AddHeader("NServiceBus.HasCallback", Boolean.TrueString)
```

Comment:
>YAGNI

Are you calling YAGNI on naming a header correctly? (remember that we have no good way to version headers so we'd have to live with it for ever)

>I don't see that header used anywhere. Am I missing something?

It will be used by the upcoming routing changes that will need to route messages with callbacks properly.

Could also be used in SI to highlight that messages had callbacks waiting for them